diff -ruN ./TorrentMonitor-master/class/TorrServer.class.php ./TorrentMonitor-master-ts/class/TorrServer.class.php
--- ./TorrentMonitor-master/class/TorrServer.class.php	1970-01-01 01:00:00.000000000 +0100
+++ ./TorrentMonitor-master-ts/class/TorrServer.class.php	2023-04-11 06:55:57.467363969 +0200
@@ -0,0 +1,87 @@
+<?php
+$dir = dirname(__FILE__).'/';
+
+class TorrServer
+{
+    #добавляем новую закачку в torrent-клиент, обновляем hash в базе
+    public static function addNew($id, $file, $hash, $tracker)
+    {
+        #получаем настройки из базы
+        $settings = Database::getAllSetting();
+        foreach ($settings as $row)
+        {
+        	extract($row);
+        }
+
+    	try
+    	{
+            if ( ! empty($hash))
+            {
+                if ($deleteDistribution)
+                {
+                    $data = 
+                        array(
+                            "action" => "drop",
+                            "hash" => $hash
+                        );
+                    $data = json_encode($data);
+                                        
+                    $request_headers[] = "Content-Type: application/json";
+                    
+                    $ch = curl_init();
+                    curl_setopt_array($ch, array(
+                        CURLOPT_POST => 1,
+                        CURLOPT_FOLLOWLOCATION => 1,
+                        CURLOPT_URL => 'http://'.$torrentAddress.'/torrents/',
+                        CURLOPT_USERAGENT => 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.12; rv:51.0) Gecko/20100101 Firefox/51.0',
+                        CURLOPT_RETURNTRANSFER => true,
+                        CURLOPT_POST => true,
+                        CURLOPT_VERBOSE => true,
+                        CURLOPT_USERPWD => $torrentLogin.':'.$torrentPassword,
+                        CURLOPT_HTTPHEADER => $request_headers,
+                        CURLOPT_POSTFIELDS => $data,
+                    ));
+                    $response = curl_exec($ch);
+                    var_dump($response);
+                    curl_close($ch);
+                }
+            }
+
+            #добавляем торрент в torrent-клиент
+            $page = Sys::getUrlContent(
+                array(
+                    'type'           => 'GET',
+                    'header'         => 1,
+                    'follow'         => 1,
+                    'returntransfer' => 1,
+                    'userpwd'        => $torrentLogin.':'.$torrentPassword,
+                    'url'            => 'http://'.$torrentAddress.'/stream/fname?link='.$file.'&save&stat',
+                )
+            );
+            
+            preg_match_all('/\"hash\":\"(.*)\"/U', $page, $res);
+            if ($res[1])
+            {
+                $hashNew = $res[1][0];
+    
+                Database::updateHash($id, $hashNew);
+                Database::clearWarnings('TorrServer');
+                
+                $return['status'] = TRUE;
+                $return['hash'] = $hashNew;
+            }
+            else
+            {
+                $return['status'] = FALSE;
+                $return['msg'] = 'unknown';
+            }
+        }
+        catch (Exception $e)
+        {
+            echo $e->getMessage().PHP_EOL;
+        }
+        
+        return $return;
+    }
+}
+?>
\ No newline at end of file
diff -ruN ./TorrentMonitor-master/include/settings.php ./TorrentMonitor-master-ts/include/settings.php
--- ./TorrentMonitor-master/include/settings.php	2023-04-07 07:56:56.000000000 +0200
+++ ./TorrentMonitor-master-ts/include/settings.php	2023-04-11 06:56:10.327611052 +0200
@@ -266,7 +266,7 @@
                     <div class="col --2:lg mb-1">Торрент-клиент:</div>
                     <div class="col --5:lg mb-2">
                         <select x-model="options.torrentClient">
-                            <template x-for='client in ["Deluge","Transmission","qBittorrent"]'>
+                            <template x-for='client in ["Deluge","Transmission","qBittorrent","TorrServer"]'>
                                 <option x-text="client" :selected="options.torrentClient == client"></option>
                             </template>
                         </select>
