diff -ru ./TorrentMonitor-master-old/trackers/kinozal.me.engine.php ./TorrentMonitor-master/trackers/kinozal.me.engine.php
--- ./TorrentMonitor-master-old/trackers/kinozal.me.engine.php	2023-02-09 18:36:26.176682244 +0300
+++ ./TorrentMonitor-master/trackers/kinozal.me.engine.php	2023-02-09 18:39:44.790768698 +0300
@@ -37,7 +37,6 @@
 	//функция преобразования даты
 	private static function dateStringToNum($data)
 	{
-		$data = "сейчас";
 	    if (strstr($data, 'сегодня') || strstr($data, 'вчера') || strstr($data, 'сейчас'))
 	    {
 	        $pieces = explode(' ', $data);
